_rules/text-contrast-afw4f7.md

```diff
@@ -0,0 +1,366 @@
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This dark grey text is on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 2
+
+This dark grey text is on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F); width: 500px;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 3
+
+This light grey text is on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:15px; background: #000 no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Passed Example 4
+
+This text passes because the text shadow gives the text sufficient contrast.
+
+```html
+<p style="color: #000; background: #737373; text-shadow: white 0 0 3px">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 5
+
+This text only has a 3:1 contrast but it is a large (18 point) text so it still passes.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 6
+
+This text only has a 3:1 contrast but it is a large (14 points and bold) text so it still passes.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+The purpose of the Helvetica letters is to show what the font looks like. The actual words are irrelevant and could be replaced with any other sentence that includes all letters of the alphabet. Because of that, the rule passes, despite the text not meeting the 4.5:1 color contrast requirement.
+
+**Note**: Because this is non-text content, [success criterion 1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG21/#non-text-contrast) requires font example to have a color contrast of 3:1.
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+This text has an insufficient contrast but it does not express anything in human language so it still passes.
+
+```html
+<p style="color: #000; background: #666;">
+	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
+</p>
+```
+
+#### Passed Example 9
+
+Even though the text is in a shadow DOM tree, the text is part of a child [text node](https://dom.spec.whatwg.org/#text) of an HTML element in the flat tree and has sufficient contrast
+
+```html
+<p style="color: #333; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
```

Comment:
Looking at the Expectation, there seems to be missing a Passed Example with decorative text.

Reply:
Passed example 7 has decorative text. We'll update the example description to make this clearer.

Reply:
I disagree.

I think that the text in Passed Example 7 (the font example) does not meet the WCAG definition of "pure decoration" that is the exception to the SC:
> serving only an aesthetic purpose, providing no information, and having no functionality

The text in that example has the functionality of showing the font. This interpretation (it is not pure decoration) is seconded by the note on that example saying that it needs a 3:1 contrast to meet 1.4.11. I agree that it qualifies as a "Graphical Object" for 1.4.11:
> Graphical Objects: Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.
(even if it's "text graphics").

In any case, if the text in that example is "pure decoration", the note and the reference to 1.4.11 need to go away. If it is not "pure decoration" (which I think), then a "pure decoration" example should be added.

Reply:
"functionality" as it is used in WCAG refers to things users can do on the page. Adding an item to a shopping cart is functionality.  See the definition of [functionality](https://www.w3.org/TR/WCAG21/#dfn-functionality). The font shape isn't "information" either, otherwise 1.3.1 would apply, information conveyed through presentation. 

Additionally, if this wasn't "purely decorative", it wouldn't be exempt under the 1.4.3 "incidental" bullet at all. Here's what 1.4.3's understanding doc says about it:

> For example, if random words are used to create a background and the words could be rearranged or substituted without changing meaning, then it would be decorative and would not need to meet this criterion.

This is precisely the same here. You can change the words without changing the meaning. That makes it decorative.

Reply:
I agree that the text is decorative.
I do not understand why a decorative text is relevant for 1.4.11 (as pointed out by the note in Passed Example 7).
(it is not a UI component, and it is not a "Graphical Object" either; so 1.4.11 does not apply)

This is not important enough to block that PR…

Reply:
Passed Example 7 is a little confusing. The first paragraph passes because it is using browser default styles. If the second paragraph is decorative, then it is an example of inapplicable.

Not sure how to split it as an example though. Perhaps a different example of each is needed.

Reply:
I'll put a foreground and background color on passed example 7 to clear this up, and add a separate example for browser defaults. That's a good point.